core/src/main/java/com/amplifyframework/predictions/models/FaceLivenessSessionInformation.kt

```diff
@@ -17,26 +17,58 @@ package com.amplifyframework.predictions.models
 import com.amplifyframework.annotations.InternalAmplifyApi
 
 @InternalAmplifyApi
-data class FaceLivenessSessionInformation(
-    val videoWidth: Float,
-    val videoHeight: Float,
-    val challenge: String? = null,
-    val region: String,
-    val challengeVersions: List<Challenge>? = null,
-    val preCheckViewEnabled: Boolean? = null,
-    val attemptCount: Int? = null
-)
+class FaceLivenessSessionInformation {
+    val videoWidth: Float
+    val videoHeight: Float
+    val challengeVersions: List<Challenge>
+    val region: String
+    val preCheckViewEnabled: Boolean?
+    val attemptCount: Int?
 
-@InternalAmplifyApi
-data class Challenge(
-    val type: ChallengeType,
-    val version: String
-) {
-    fun toQueryParamString(): String = "${type.name}_$version"
+    @Deprecated("Keeping compatibility for <= Amplify Liveness 1.2.6")
+    constructor(
+        videoWidth: Float,
+        videoHeight: Float,
+        challenge: String,
+        region: String
+    ) {
+        this.videoWidth = videoWidth
+        this.videoHeight = videoHeight
+        this.challengeVersions = listOf(Challenge.FaceMovementAndLightChallenge("1.0.0"))
+        this.region = region
+        this.preCheckViewEnabled = null
+        this.attemptCount = null
+    }
+
+    constructor(
+        videoWidth: Float,
+        videoHeight: Float,
+        region: String,
+        challengeVersions: List<Challenge>,
+        preCheckViewEnabled: Boolean,
+        attemptCount: Int
+    ) {
+        this.videoWidth = videoWidth
+        this.videoHeight = videoHeight
+        this.region = region
+        this.challengeVersions = challengeVersions
+        this.preCheckViewEnabled = preCheckViewEnabled
+        this.attemptCount = attemptCount
+    }
 }
 
 @InternalAmplifyApi
-enum class ChallengeType {
-    FaceMovementChallenge,
-    FaceMovementAndLightChallenge,
+sealed class Challenge private constructor(val name: String, val version: String) {
+
+    @InternalAmplifyApi
+    class FaceMovementChallenge(version: String) : Challenge("FaceMovementChallenge", version)
+
+    @InternalAmplifyApi
+    class FaceMovementAndLightChallenge(version: String) : Challenge("FaceMovementAndLightChallenge", version)
+
+    fun compareType(challenge: Challenge): Boolean {
```

Comment:
Can't you just make these data classes and then you can just do an `==` check

Reply:
You know I never thought to use data classes in sealed classes lol yes I've updated this